// @shopify/cli/dist/hooks/postrun.js
import {
  getNextDeprecationDate
} from "../chunk-NPH2SXRO.js";
import {
  reportAnalyticsEvent
} from "../chunk-UBCH575K.js";
import "../chunk-DL3MKXVR.js";
import "../chunk-O3JOUAA5.js";
import "../chunk-N3YORLAS.js";
import "../chunk-3LDWVYMD.js";
import "../chunk-3GSKXZGY.js";
import "../chunk-PRJ3U7MR.js";
import {
  addSensitiveMetadata,
  getAllSensitiveMetadata,
  outputDebug,
  renderWarning
} from "../chunk-T4Y7NDNJ.js";
import "../chunk-JIJ65JT6.js";
import "../chunk-UMWKJFTM.js";
import "../chunk-GEN4RXTD.js";
import "../chunk-BYC3NA2Y.js";
import "../chunk-SAR5HGRG.js";
import "../chunk-NABCTJNW.js";
import "../chunk-YC4FLMDD.js";
import "../chunk-QNY4RVJ6.js";
import "../chunk-LSCXQWNO.js";
import "../chunk-K6257PKA.js";
import "../chunk-XQFDS44U.js";
import "../chunk-AMAYT2P2.js";
import "../chunk-VM4POVNV.js";
import "../chunk-SWNRLCC6.js";
import "../chunk-T5LLJYYS.js";
import {
  init_cjs_shims
} from "../chunk-ZPL24Y2D.js";

// src/hooks/postrun.ts
init_cjs_shims();

// ../cli-kit/dist/public/node/hooks/postrun.js
init_cjs_shims();

// ../cli-kit/dist/public/node/hooks/deprecations.js
init_cjs_shims();
var postrun = (Command) => {
  const nextDeprecationDate = getNextDeprecationDate();
  if (nextDeprecationDate) {
    const forThemes = Command?.id?.includes("theme");
    renderUpgradeWarning(nextDeprecationDate, forThemes);
  }
};
function renderUpgradeWarning(upgradeByDate, forThemes) {
  const dateFormat = new Intl.DateTimeFormat("default", {
    year: "numeric",
    month: "long",
    day: "numeric"
  });
  const formattedDate = dateFormat.format(upgradeByDate);
  const headline = `Upgrade to the latest CLI version by ${formattedDate}.`;
  const body = "This command requires an upgrade to continue working as intended.";
  const upgradeLink = {
    link: {
      label: "upgrade Shopify CLI",
      url: forThemes ? `https://shopify.dev/docs/themes/tools/cli#upgrade-shopify-cli` : `https://shopify.dev/docs/apps/tools/cli#upgrade-shopify-cli`
    }
  };
  const nextSteps = [["Run", { command: "upgrade" }, "to", upgradeLink]];
  renderWarning({
    headline,
    body,
    nextSteps
  });
}

// ../cli-kit/dist/public/node/hooks/postrun.js
var hook = async ({ config, Command }) => {
  await detectStopCommand(Command);
  await reportAnalyticsEvent({ config, exitMode: "ok" });
  postrun(Command);
  const command = Command?.id?.replace(/:/g, " ");
  outputDebug(`Completed command ${command}`);
};
async function detectStopCommand(commandClass) {
  const currentTime = (/* @__PURE__ */ new Date()).getTime();
  if (commandClass && Object.prototype.hasOwnProperty.call(commandClass, "analyticsStopCommand")) {
    const stopCommand = commandClass.analyticsStopCommand();
    if (stopCommand) {
      const { commandStartOptions } = getAllSensitiveMetadata();
      await addSensitiveMetadata(() => ({
        commandStartOptions: {
          ...commandStartOptions,
          startTime: currentTime,
          startCommand: stopCommand
        }
      }));
    }
  }
}
export {
  hook as default
};
