// @shopify/cli/dist/chunk-R5GT4GBL.js
import {
  demo
} from "./chunk-LAJ4OEME.js";
import {
  base_command_default
} from "./chunk-PEAIOYXD.js";
import {
  readFile
} from "./chunk-T4Y7NDNJ.js";
import {
  require_lib
} from "./chunk-GEN4RXTD.js";
import {
  cwd,
  resolvePath
} from "./chunk-QNY4RVJ6.js";
import {
  __toESM,
  init_cjs_shims
} from "./chunk-ZPL24Y2D.js";

// src/cli/commands/demo/index.ts
init_cjs_shims();
var import_core = __toESM(require_lib(), 1);
var Demo = class _Demo extends base_command_default {
  static {
    this.description = "Demo a command design defined in a file";
  }
  static {
    this.hidden = true;
  }
  static {
    this.flags = {
      path: import_core.Flags.string({
        hidden: false,
        description: "The directory where the demo file is located. Defaults to the current directory.",
        env: "SHOPIFY_FLAG_PATH",
        parse: async (input) => resolvePath(input),
        default: async () => cwd()
      }),
      file: import_core.Flags.string({
        hidden: false,
        description: "The name of the demo file.",
        env: "SHOPIFY_FLAG_PATH",
        parse: async (input) => resolvePath(input),
        default: async () => cwd()
      })
    };
  }
  async run() {
    const { flags } = await this.parse(_Demo);
    const contents = await readFile(flags.file);
    const design = JSON.parse(contents);
    await demo(design);
  }
};

export {
  Demo
};
