// @shopify/cli/dist/local-OQXN5NM2.js
import {
  alwaysLogAnalytics,
  alwaysLogMetrics,
  analyticsDisabled,
  ciPlatform,
  cloudEnvironment,
  codespacePortForwardingDomain,
  codespaceURL,
  firstPartyDev,
  gitpodURL,
  hasGit,
  homeDirectory,
  isCloudEnvironment,
  isDevelopment,
  isShopify,
  isTerminalInteractive,
  isUnitTest,
  isVerbose,
  macAddress,
  opentelemetryDomain,
  useDeviceAuth,
  useEmbeddedThemeCLI,
  useThemebundling
} from "./chunk-T4Y7NDNJ.js";
import "./chunk-JIJ65JT6.js";
import "./chunk-UMWKJFTM.js";
import "./chunk-GEN4RXTD.js";
import "./chunk-BYC3NA2Y.js";
import "./chunk-SAR5HGRG.js";
import "./chunk-NABCTJNW.js";
import "./chunk-YC4FLMDD.js";
import "./chunk-QNY4RVJ6.js";
import "./chunk-LSCXQWNO.js";
import "./chunk-K6257PKA.js";
import "./chunk-XQFDS44U.js";
import "./chunk-AMAYT2P2.js";
import "./chunk-VM4POVNV.js";
import "./chunk-SWNRLCC6.js";
import "./chunk-T5LLJYYS.js";
import "./chunk-ZPL24Y2D.js";
export {
  alwaysLogAnalytics,
  alwaysLogMetrics,
  analyticsDisabled,
  ciPlatform,
  cloudEnvironment,
  codespacePortForwardingDomain,
  codespaceURL,
  firstPartyDev,
  gitpodURL,
  hasGit,
  homeDirectory,
  isCloudEnvironment,
  isDevelopment,
  isShopify,
  isTerminalInteractive,
  isUnitTest,
  isVerbose,
  macAddress,
  opentelemetryDomain,
  useDeviceAuth,
  useEmbeddedThemeCLI,
  useThemebundling
};
