// @shopify/cli/dist/workerd-OLKE7G4X.js
import {
  H2O_BINDING_NAME,
  SUBREQUEST_PROFILER_ENDPOINT,
  TUNNEL_DOMAIN,
  createLogRequestEvent,
  getDebugBannerLine,
  getUtilityBannerlines,
  handleDebugNetworkRequest,
  handleMiniOxygenImportFail,
  importLocal,
  logRequestLine,
  setConstructors
} from "./chunk-K7KD247K.js";
import "./chunk-U2PN6QZ2.js";
import "./chunk-NPH2SXRO.js";
import "./chunk-EJITPGUJ.js";
import "./chunk-PEAIOYXD.js";
import "./chunk-EIUQV76I.js";
import "./chunk-UBCH575K.js";
import "./chunk-DL3MKXVR.js";
import "./chunk-O3JOUAA5.js";
import "./chunk-N3YORLAS.js";
import "./chunk-FMKVOTQK.js";
import "./chunk-3LDWVYMD.js";
import "./chunk-3GSKXZGY.js";
import "./chunk-PRJ3U7MR.js";
import {
  AbortError,
  createFileReadStream,
  outputNewline,
  readFile,
  renderSuccess,
  source_default
} from "./chunk-T4Y7NDNJ.js";
import "./chunk-JIJ65JT6.js";
import "./chunk-UMWKJFTM.js";
import "./chunk-GEN4RXTD.js";
import "./chunk-BYC3NA2Y.js";
import "./chunk-SAR5HGRG.js";
import "./chunk-NABCTJNW.js";
import "./chunk-YC4FLMDD.js";
import {
  dirname,
  resolvePath
} from "./chunk-QNY4RVJ6.js";
import "./chunk-LSCXQWNO.js";
import "./chunk-K6257PKA.js";
import "./chunk-XQFDS44U.js";
import "./chunk-AMAYT2P2.js";
import "./chunk-VM4POVNV.js";
import "./chunk-SWNRLCC6.js";
import "./chunk-T5LLJYYS.js";
import {
  init_cjs_shims
} from "./chunk-ZPL24Y2D.js";

// ../../node_modules/.pnpm/@shopify+cli-hydrogen@8.4.0_@graphql-codegen+cli@5.0.2_react-dom@17.0.2_react@17.0.2/node_modules/@shopify/cli-hydrogen/dist/lib/mini-oxygen/workerd.js
init_cjs_shims();
import { createRequire } from "node:module";
async function startWorkerdServer({
  root,
  appPort,
  inspectorPort: publicInspectorPort,
  assetsPort,
  debug = false,
  watch = false,
  buildPathWorkerFile,
  buildPathClient,
  env
}) {
  const { createMiniOxygen, Response } = await importLocal(
    "@shopify/mini-oxygen",
    root
  ).catch(handleMiniOxygenImportFail);
  setConstructors({ Response });
  async function handleCustomerAccountSchema() {
    const require2 = createRequire(import.meta.url);
    const filePath = require2.resolve(
      "@shopify/hydrogen/customer-account.schema.json"
    );
    return new Response(createFileReadStream(filePath), {
      headers: { "Content-Type": "application/json" }
    });
  }
  const mainWorkerName = "hydrogen";
  const absoluteBundlePath = resolvePath(root, buildPathWorkerFile);
  const readWorkerFile = () => readFile(absoluteBundlePath).catch((error) => {
    throw new AbortError(
      `Could not read worker file.

` + error.stack,
      "Did you build the project?"
    );
  });
  const miniOxygen = createMiniOxygen({
    debug,
    port: appPort,
    host: "localhost",
    liveReload: watch,
    requestHook: logRequestLine,
    inspectorPort: publicInspectorPort,
    inspectWorkerName: mainWorkerName,
    assets: { port: assetsPort, directory: buildPathClient },
    workers: [
      {
        name: "hydrogen:middleware",
        modules: true,
        script: `export default { fetch: (request, env) => {
          const url = new URL(request.url);
          if (url.hostname.endsWith('${TUNNEL_DOMAIN.ORIGINAL}')) {
            url.hostname = url.hostname.replace(
              '${TUNNEL_DOMAIN.ORIGINAL}',
              '${TUNNEL_DOMAIN.REBRANDED}',
            );
          }

          return url.pathname === '${SUBREQUEST_PROFILER_ENDPOINT}'
            ? env.profiler.fetch(url, request)
            : url.pathname === '/graphiql/customer-account.schema.json'
            ? env.assets.fetch(url, request)
            : env.next.fetch(url, request)
          }
        }`,
        serviceBindings: {
          profiler: handleDebugNetworkRequest,
          assets: handleCustomerAccountSchema,
          next: mainWorkerName
        }
      },
      {
        name: mainWorkerName,
        modulesRoot: dirname(absoluteBundlePath),
        modules: [
          {
            type: "ESModule",
            path: absoluteBundlePath,
            contents: await readWorkerFile()
          }
        ],
        bindings: { ...env },
        serviceBindings: {
          [H2O_BINDING_NAME]: createLogRequestEvent({
            transformLocation: () => absoluteBundlePath
          })
        }
      }
    ]
  });
  const { workerUrl, inspectorUrl } = await miniOxygen.ready;
  return {
    port: Number(workerUrl.port),
    listeningAt: workerUrl.origin,
    reload(nextOptions) {
      return miniOxygen.reload(async ({ workers }) => {
        const mainWorker = workers.find(({ name }) => name === mainWorkerName);
        if (Array.isArray(mainWorker.modules) && mainWorker.modules[0]) {
          mainWorker.modules[0].contents = await readWorkerFile();
        }
        if (nextOptions) {
          mainWorker.bindings = { ...nextOptions?.env ?? env };
        }
        return { workers };
      });
    },
    showBanner(options) {
      outputNewline();
      const customSections = [];
      if (options?.host) {
        customSections.push({ body: getUtilityBannerlines(options.host) });
      }
      if (inspectorUrl) {
        customSections.push({
          body: { warn: getDebugBannerLine(Number(inspectorUrl.port)) }
        });
      }
      renderSuccess({
        headline: `${options?.headlinePrefix ?? ""}MiniOxygen (Worker Runtime) ${options?.mode ?? "development"} server running.`,
        body: [
          `View ${options?.appName ? source_default.cyan(options?.appName) : "Hydrogen"} app:`,
          { link: { url: options?.host || workerUrl.origin } }
        ],
        customSections
      });
      console.log("");
    },
    async close() {
      await miniOxygen.dispose();
    }
  };
}
export {
  startWorkerdServer
};
